### ISSUE TITLE
Broken link in app icon after upgrade from 8.2 to 9.0

### ISSUE THREAD

**Issue by rseek** [owncloud/music#478]
Hello!
I am using music app 0.3.10

After upgrade of OC from 8.2 to 9.0 on my ubuntu 14.04, I faced really strange behavior of music app:
When I click on music app icon in navigation drop-down menu, I just have my page reloaded (or loaded again - if there is such a difference)

When inspected page source code, I found out that there is an empty link in music app section, while in other app sections there are proper links:
```
<a href="" tabindex="3">
							<img class="app-icon svg" alt="" src="/apps/music/img/music.svg">
							<div class="icon-loading-dark" style="display:none;"></div>
							<span>
								Music							</span>
						</a>
```

**Comment by susca** [comment]
Hi,

I can confirm this. If I manually correct the URL in my browsers address bar (my.server.org/owncloud/index.php/apps/music), I get the app to run.

**Comment by susca** [comment]
I noticed that I get this error in the log:

Exception:` {"Exception":"Symfony\\Component\\Routing\\Exception\\RouteNotFoundException","Message":"Unable to generate a URL for the named route \"music.page.index\" as such route does not exist.","Code":0,"Trace":"#0 \/var\/www\/owncloud\/lib\/private\/route\/router.php(313): Symfony\\Component\\Routing\\Generator\\UrlGenerator->generate('music.page.inde...', Array, 1)\n#1 \/var\/www\/owncloud\/lib\/private\/urlgenerator.php(66): OC\\Route\\Router->generate('music.page.inde...', Array)\n#2 \/var\/www\/owncloud\/apps\/music\/appinfo\/app.php(26): OC\\URLGenerator->linkToRoute('music.page.inde...')\n#3 \/var\/www\/owncloud\/lib\/private\/app.php(163): require_once('\/var\/www\/ownclo...')\n#4 \/var\/www\/owncloud\/lib\/private\/app.php(144): OC_App::requireAppFile('music')\n#5 \/var\/www\/owncloud\/lib\/private\/app.php(117): OC_App::loadApp('music')\n#6 \/var\/www\/owncloud\/lib\/private\/util.php(137): OC_App::loadApps(Array)\n#7 \/var\/www\/owncloud\/lib\/private\/files\/filesystem.php(242): OC_Util::setupFS(NULL)\n#8 \/var\/www\/owncloud\/lib\/private\/server.php(166): OC\\Files\\Filesystem::getMountManager(NULL)\n#9 \/var\/www\/owncloud\/3rdparty\/pimple\/pimple\/src\/Pimple\/Container.php(113): OC\\Server->OC\\{closure}(Object(OC\\Server))\n#10 \/var\/www\/owncloud\/lib\/private\/appframework\/utility\/simplecontainer.php(102): Pimple\\Container->offsetGet('RootFolder')\n#11 \/var\/www\/owncloud\/lib\/private\/servercontainer.php(87): OC\\AppFramework\\Utility\\SimpleContainer->query('RootFolder')\n#12 \/var\/www\/owncloud\/lib\/private\/server.php(720): OC\\ServerContainer->query('RootFolder')\n#13 \/var\/www\/owncloud\/lib\/private\/server.php(617): OC\\Server->getRootFolder()\n#14 \/var\/www\/owncloud\/3rdparty\/pimple\/pimple\/src\/Pimple\/Container.php(113): OC\\Server->OC\\{closure}(Object(OC\\Server))\n#15 \/var\/www\/owncloud\/lib\/private\/appframework\/utility\/simplecontainer.php(102): Pimple\\Container->offsetGet('ShareManager')\n#16 \/var\/www\/owncloud\/lib\/private\/servercontainer.php(87): OC\\AppFramework\\Utility\\SimpleContainer->query('ShareManager')\n#17 \/var\/www\/owncloud\/lib\/private\/server.php(1272): OC\\ServerContainer->query('ShareManager')\n#18 \/var\/www\/owncloud\/apps\/files_sharing\/appinfo\/routes.php(130): OC\\Server->getShareManager()\n#19 \/var\/www\/owncloud\/lib\/private\/route\/router.php(327): include_once('\/var\/www\/ownclo...')\n#20 \/var\/www\/owncloud\/lib\/private\/route\/router.php(142): OC\\Route\\Router->requireRouteFile('\/var\/www\/ownclo...', 'files_sharing')\n#21 \/var\/www\/owncloud\/lib\/private\/route\/router.php(307): OC\\Route\\Router->loadRoutes()\n#22 \/var\/www\/owncloud\/lib\/private\/urlgenerator.php(66): OC\\Route\\Router->generate('mail.page.index', Array)\n#23 \/var\/www\/owncloud\/apps\/mail\/appinfo\/app.php(40): OC\\URLGenerator->linkToRoute('mail.page.index')\n#24 \/var\/www\/owncloud\/lib\/private\/app.php(163): require_once('\/var\/www\/ownclo...')\n#25 \/var\/www\/owncloud\/lib\/private\/app.php(144): OC_App::requireAppFile('mail')\n#26 \/var\/www\/owncloud\/lib\/private\/app.php(117): OC_App::loadApp('mail')\n#27 \/var\/www\/owncloud\/lib\/base.php(862): OC_App::loadApps()\n#28 \/var\/www\/owncloud\/index.php(39): OC::handleRequest()\n#29 {main}","File":"\/var\/www\/owncloud\/3rdparty\/symfony\/routing\/Generator\/UrlGenerator.php","Line":130} 

Maybe it helps...

**Comment by MorrisJobke** [comment]
I can't reproduce this anymore - even with 0.3.10 on stable9 - let me test with a 9.0.0 owncloud.

**Comment by MorrisJobke** [comment]
I tested both: with and without the shortened URL syntax (with and without the `index.php`) on 9.0.0. All works fine.

@benediktg Do you maybe have spotted more? Any ideas?

**Comment by susca** [comment]
@MorrisJobke: Problem is fixed for me (OC 9.0.1) after update to music 0.3.11 . Thank you.

**Comment by MorrisJobke** [comment]
> @MorrisJobke: Problem is fixed for me (OC 9.0.1) after update to music 0.3.11 . Thank you.

Awesome 🎉
